library/core/resource_loader/src/main/java/org/quiltmc/qsl/resource/loader/api/QuiltResourcePack.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2021-2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.resource.loader.api;
+
+import net.minecraft.resource.pack.ResourcePack;
+import net.minecraft.text.Text;
+
+import org.quiltmc.qsl.base.api.util.InjectedInterface;
+
+/**
+ * Represents a resource pack with extended metadata, injected into {@link net.minecraft.resource.pack.ResourcePack}.
+ */
+@InjectedInterface(ResourcePack.class)
+public interface QuiltResourcePack {
+	/**
+	 * {@return a display name for this resource pack}
+	 */
+	Text getDisplayName();
```

Comment:
Why is this not `default`?

Reply:
This is an injected interface, the default implementation is in the mixin.
If a `default` was provided it wouldn't work well.

Reply:
Yes.

Not sure what the proper path should be.
Should I add a simple default dummy return value?